metriks/distribution_test.go

```diff
@@ -0,0 +1,35 @@
+package metriks
+
+import (
+	"bytes"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDistribution(t *testing.T) {
+	pc, err := net.ListenPacket("udp", ":10000")
+	require.NoError(t, err)
+	defer pc.Close()
+
+	msgs := make(chan []byte)
+	go func() {
+		buf := make([]byte, 1024)
+		_, _, err := pc.ReadFrom(buf)
+		require.NoError(t, err)
+		msgs <- buf
+		close(msgs)
+	}()
+
+	require.NoError(t, initDistribution(pc.LocalAddr().String(), "testing", []string{"a:1"}))
+	SetDistributionErrorHandler(func(err error) {
+		assert.NoError(t, err)
+	})
+
+	Distribution("some_metric", 12.0, L("b", "c"))
+
+	buf := <-msgs
+	assert.Equal(t, "some_metric.some_metric:12|d|#a:1,b:c", string(bytes.Trim(buf, "\x00")))
```

Comment:
Does this add the service name prefix if constructed from the `metriks.Init` pkg?

Reply:
arg ~ i messed up the variable scoping thing. Should be fixed

Reply:
the actual prefixing happens deep in the go-metrics statsd sink